Enabled running lv-web without docker via npm

Mohase opened the pull request:
## What
Enabled running lv-web with npm instead of Docker for faster (front end) development.

## Changes
- Added `--` to `dev:lv-web` script to pass CLI arguments
- Added documentation for running without Docker (README.md)

## How to use (README)
1. Create `apps/lv-web/.env.local` with localhost database URL
2. Start database: `docker compose --profile lv-web up db -d`
3. Run app: `npm run dev:lv-web -- --port=3045`

## Tested locally 
 Next.js starts on port 3045
 Sign-in works with database in Docker

Mohase committed:
Enabled running lv-web without docker via npm

hilalnurgercek commented:
A few questions from my side:
- We already had an .env file in lv-web, what is the reason for creating also .env.local? Or would .env be enough on its own?
- I initially had port 5432 for `DATABASE_URL` and `DIRECT_DATABASE_URL,` which was set I think in the beginning of the project? Is there a reason for switching to port 3772 in the URLs?
- After I ran the commands specified in README one by one, `npm run dev:lv-web -- --port=3045` gave me:
```
> galactic@0.1.0 dev:lv-web
> NEXT_PUBLIC_APP_SOURCE_NAME=lv-web turbo run dev --filter=lv-web --port=3045
 ERROR  unexpected argument '--port' found
  tip: to pass '--port' as a value, use '-- --port'
Usage: turbo run <--cache-dir <CACHE_DIR>|--concurrency <CONCURRENCY>|--continue[=<CONTINUE>]|--single-package|--framework-inference [<BOOL>]|--global-deps <GLOBAL_DEPS>|--env-mode [<ENV_MODE>]|--filter <FILTER>|--affected|--output-logs <OUTPUT_LOGS>|--log-order <LOG_ORDER>|--only|--pkg-inference-root <PKG_INFERENCE_ROOT>|--log-prefix <LOG_PREFIX>|TASKS|PASS_THROUGH_ARGS> <--filter <FILTER>|--affected>
For more information, try '--help'.
```
Any ideas on how to resolve this? I'm on Mac, I don't know if that's causing some problems :/

Mohase commented:
The `--` tells npm to pass `--port=3045` to Next.js (not to turbo). Without it, turbo tries to parse `--port` as its own flag. Im not sure why you had the error because the script already includes `--`: "dev:lv-web": "NEXT_PUBLIC_APP_SOURCE_NAME=lv-web turbo run dev --filter=lv-web --". See the one after "web"? 

The problem most likely is that you didn't run the docker compose --profile lv-web up db -d and  npm run dev:lv-web -- --port=3045 commands from the root, this was my mistake because I removed the cd ../../.. from read me and didn't specify it must be run from project root. I'll update readme and commit. Also if you already had done that you might have not pulled the latest version. 

Regarding `.env.local` vs `.env`:
`.env.local` is required, not optional. When I tested, `.env` didn't work. Next.js prioritizes `.env.local` for local development, and it's what the PO recommended in the issue. `.env.local` is also gitignored by default, which is safer for local config. This issue was done so front end editing and such would be easier for the developer and they won't have to build the whole program every time. That way you can build it using local env.

Regarding port 3772 vs 5432:
Port 3772 is the Docker-exposed port (db service maps 3772:5432). When running npm outside Docker, you need 3772 to reach the containerized database. I might be wrong but port 5432 would only work if you had a local Postgres instance running, or if the Docker mapping was different. Im not exactly sure what you mean but it works like this 5432 for env and 3772 for env.local. I checked manually whether it would work with db:5432 and localhost:5432, and it does not.

Mohase committed:
Merge branch 'dev' into mohammad/enable-npm-without-docker

hilalnurgercek commented:
I’ve managed to get past the error and can log in now after some debugging. However, the Python API status in the dashboard is showing as off. Are you encountering the same issue?
<img width="365" height="311" alt="image" src="https://github.com/user-attachments/assets/050d28dd-e807-49e0-98ca-84e20c7eeedf" />

CatSpectre reviewed (commented):
I had to delete `apps/lv-web/.next` like suggested in the issue before `npm run dev:lv-web -- --port=3045` did anything. Well after that and all the other steps mentioned I cant sign in and wsl said bunch of stuff but mainly this:
```
lv-web:dev: https://next-auth.js.org/errors#adapter_error_getuserbyaccount
lv-web:dev: Invalid `prisma.account.findUnique()` invocation:
lv-web:dev:
lv-web:dev:
lv-web:dev: The table `public.Account` does not exist in the current database.
```

Mohase commented:
> I had to delete `apps/lv-web/.next` like suggested in the issue before `npm run dev:lv-web -- --port=3045` did anything. Well after that and all the other steps mentioned I cant sign in and wsl said bunch of stuff but mainly this:
> 
> ```
> lv-web:dev: https://next-auth.js.org/errors#adapter_error_getuserbyaccount
> lv-web:dev: Invalid `prisma.account.findUnique()` invocation:
> lv-web:dev:
> lv-web:dev:
> lv-web:dev: The table `public.Account` does not exist in the current database.
> ```

im not sure, why it doesnt work with WSL because I don't have windows so I can't really check what works or not, but after discussing with Hilal I tested the program for another hour and the way it works in this pull request is the only way I got it to work. Switching ports and what not didn't work for me and like I said I can't test if it works for windows. Perhaps you could edit it so it works on windows so we can merge.

Mohase committed:
Merge branch 'dev' into mohammad/enable-npm-without-docker

johottaja commented:
It looks like you deleted the testing part of README.md??

Mohase committed:
Fixed README: Restoreed missing Testing section

- Added Testing section with instructions for running and writing tests
- Placed after Test Builds Locally section for logical flow
- Includes Jest and React Testing Library documentation

Mohase commented:
> It looks like you deleted the testing part of README.md??

should be fixed now

Aaryan-nyr commented:
Was able to build and access UI on Windows (without wsl)

Aaryan-nyr reviewed (approved):
Works on Windows

CatSpectre reviewed (commented):
I got it finally working fully. Some things:
- Lv-web's `.env.local` needs `NEXT_PUBLIC_PYAPI_URL=http://localhost:8091` to get PYAPI connected, this is only included in docker compose file atm. Should add this to the `.env.example` or README. 
- Include `sudo rm -rf apps/lv-web/.next` in README, at least I had to do that. Also I think regenerating Prisma is required for all machines so you could remove the 'for Mac' mention.
- I think you added back wrong testing info to README, coverage stuff is missing

johottaja commented:
> Regarding `.env.local` vs `.env`:
> `.env.local` is required, not optional. When I tested, `.env` didn't work. Next.js prioritizes `.env.local` for local development, and it's what the PO recommended in the issue.

.env.local is not required. Next.js does prioritise .env.local, but that means that if you have the same variable name in both, nextjs uses the one found in .env.local instead if .env

Problem is that we need db:5432 running on the container and 3772 running on the rpm run version. It's a clever hack to have .env.local contain the database urls using localhost:3772 and .env contain db:5432 because the local nextjs build will use the urls from .env.local due to the prioritisation, and the docker build just copies all variables form .env
This strikes me as the wrong way to do it but if it's not broken, don't fix it.

Also, I had to add `binaryTargets = ["native", "darwin-arm64"]` to `packages/database/prisma/schema/prisma.schema` and the pyapi url of course. 

Everything worked for me, but maybe we should update README.md to contain all this information, also about having to run npx prisma generate in `packages/database/prisma` when switching from docker-run to npm-run.

johottaja reviewed (dismissed):
See my comment above.

Mohase commented:
also about having to run npx prisma generate in packages/database/prisma when switching from docker-run to npm-run. -> 
cd packages/database/prisma && npx prisma generate
cd ../../.. 

This line is already taking the user to the correct directory to run the generate command. The purpose was for the user to follow the steps exactly. About the local.env. As I discussed in detail with @hilalnurgercek , I tried many ways to get it to work without the local.env. but it wouldn't work. Adding them in local.env was also PO's suggestion. I'll make the fixes and merge.

Mohase committed:
Merge dev: resolve README conflicts and add PR suggestions

- Resolved merge conflicts in README.md
- Added binaryTargets to Prisma schema for darwin-arm64 support
- Updated README with .env.local vs .env explanation
- Added note about running npx prisma generate when switching between docker-run and npm-run

Mohase committed:
Restore Testing section in README

Mohase committed:
Merge branch 'dev' into mohammad/enable-npm-without-docker

Mohase committed:
Merge branch 'mohammad/enable-npm-without-docker' of https://github.com/clarvo-ai/Living-Vectors into mohammad/enable-npm-without-docker

Mohase committed:
Add NEXT_PUBLIC_PYAPI_URL and .next cleanup to README

- Added NEXT_PUBLIC_PYAPI_URL to .env.local example for npm-run setup
- Added step to clean .next cache when switching from Docker
- Kept 'for your platform' wording (not Mac-specific)
